dhall/src/Dhall/TypeCheck.hs

```diff
@@ -546,19 +527,15 @@ typeWithA tpa = loop
             _          -> Left (TypeError ctx e (MustCombineARecord '∧' kvsY tKvsY))
 
         ttKvsX <- fmap Dhall.Core.normalize (loop ctx tKvsX)
-        constX <- case ttKvsX of
+        _      <- case ttKvsX of
             Const constX -> return constX
```

Comment:
Just return `()`.

Reply:
Or is this check necessary at all?

Reply:
It is not necessary any longer, so I'll remove it.  One invariant of the type system is that the type of the type of something will always be a type-checking constant (i.e. `Const`) so if we don't care about the specific constant then we no longer need to check this